src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,780 @@
+                } else if (type == 'D') {
+                    int i = des.indexOf('(');
+                    newTask = new Deadline(des.substring(0, i - 1),
+                            parser.parseStringToDate(des.substring(i + 5, i + 15)));
+                } else if (type == 'E') {
+                    int i = des.indexOf('(');
+                    newTask = new Event(des.substring(0, i - 1),
+                            parser.parseStringToDate(des.substring(i + 5, i + 15)));
+                } else {
+                    newTask = null;
+                }
+
+                taskList.addTask(newTask);
+            }
+        }
+
+        /**
+         * Write data to the text file "Duke.txt".
+         */
+        public void saveData() {
+            try {
+                FileWriter fw = new FileWriter(filePath);
+                String s = "";
+                for (int i = 0; i < all.size(); i++) {
+                    s += all.get(i).toString() + "\n";
+                }
+                fw.write(s);
+                fw.close();
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+    }
+
+    private abstract class Command {
+        /**
+         * Execute the command based on its type.
+         */
+        public abstract void execute();
+
+        /**
+         * Returns whether it is an exit command.
+         *
+         * @return a boolean of whether it is an exit command.
+         */
+        public abstract boolean isExit();
+    }
+
+    private class AddCommand extends Command {
+        private TaskType type;
+        private String description;
+        private LocalDate date;
+
+        public AddCommand(TaskType type, String description ) {
+            super();
+            this.type = type;
+            this.description = description;
+        }
+
+
+        /**
+         * Execute the add command. Determine the type of task added then add it to taskList.
+         */
+        @Override
+        public void execute() {
+            if (type == TaskType.TODO) {
+                taskList.addTask(new Todo(description));
+            } else if (type == TaskType.DEADLINE) {
+                try {
+                    taskList.addTask(new Deadline(parser.extractDescription(description),
+                                                  parser.extractDate(description)));
+                } catch (DukeException e) {
+                    ui.showError("format incorrect!");
+                }
+            } else if (type == TaskType.EVENT) {
+                try {
+                    taskList.addTask(new Event(parser.extractDescription(description),
+                            parser.extractDate(description)));
+                } catch (DukeException e) {
+                    ui.showError("format incorrect!");
+                }
+            } else {
+                ui.showError("type not supported");
+            }
+        }
```

Comment:
Perhaps use of switch statements instead of if-else statements can make this code look a lot cleaner?